### DISCUSSION TITLE
Constant "To connect with sensor" notifications

### DISCUSSION THREAD

**Discussion by jcarrollpe** [j-kaltes/Juggluco#123] [Q&A] [answered]
Hi, I am new to Juggluco, looking for a permanent replacement for the FSL3 app which I have been using with xDrip for the last 13 days. I am in the US, using a rooted Pixel 7 Pro, NFC enabled and (US)Libre3 sensor. Today I uninstalled FSL3, started the sensor with Juggluco and my Librereview account credentials and connected xDrip. The app works ok and data is being shared with xDrip which is working normally as the slave. The problem is I get frequent, almost constant notifications from Juggluco "To connect with sensor".  This happens even when BG data is flowing and I can see the data is there. At first I thought it was losing connection each time and have been re-scanning until I realized most times, I am already connected to the sensor.  Is there a way to lessen the sensitivity to lost connections ? I haven't tried turbo yet due to the potential extra battery usage.  To be fair I have seen a few failures and a "status 133" but these failures are nowhere near the number of times I've checked and the app is still running and sharing data with xDrip.  What is the exact meaning of this notification? because there is no matching notification under app settings.  eg. none of them indicates a lost sensor connection. I will check out turbo, but was hoping to get a battery usage baseline before experimenting with it. Thanks. 

Edit: since the time I wrote this, I have received a legit failure, a status=19 and several NFC scan errors as well, one specific to the FSL3 sensor and one unrecognized.
Edit2: Again, I am Magisk rooted but did not consider having to hide Juggluco. I definitely understand having to hide the ex-FSL3 app, but never thought I would have to hide Juggluco... as a test.  I also re-selected "glucose notification" which I had disabled and the issue seems to have stabilized.  Maybe the glucose notification toggle in settings is replaced by "to connect with sensor" if de-selected?  I think this is where the issue is. Thankful it is behaving better. Cheers.
Edit 3:  The issue may be with that glucose notification toggle. Since I switched it back on I don't see the constant disconnects and notifications. Silence is golden.

**Comment by jcarrollpe** [comment]
Well, it was one of three things. The Android checkbox was on, now switched to off. The Glucose notification box was off, now switched on, or hiding Juggluco from Magisk. I don't know which one made the difference but it just ran all night with no alarms or other noises (reconnect, etc.). Graphs look good, and xDrip did not miss a beat with an unbroken data line all night.

**Comment by j-kaltes** [comment, marked as answer]
How did you use the FSL3 app with xDrip? Is that possible?
Android will kill apps running in the background unless they have a foreground service with a notification. For this Juggluco shows a notification. The text “To connect with sensor” is displayed to tell that Juggluco needs the foreground service to connect with a sensor. If “Glucose notification” is turned on in Left menu→Settings, Juggluco will display in the notification the glucose level plus trend arrow when a glucose value arrives. When it doesn’t receive a glucose value for 3 minutes, it says that it didn’t receive a glucose value since the time of the last glucose value.
To determine whether glucose values are coming in, you can better see whether a glucose value is displayed at the right of the screen in Juggluco. The coloring of the sensor id, show how long ago it received the last value. No coloring, means now. Totally colored means 3 minutes ago. You can also look in Left menu→Sensor for the last successful glucose value.
Normally you don’t need to re-scan Libre 3 sensors, except when you switch back from connecting Abbott’s Libre 3 app with the sensor. Scan errors don’t matter as long as the last scan was successful.
What xDrip shows is not what Juggluco receives from the sensor. xDrip transforms the data, it receives every minutes from Juggluco, to one point in 5 minutes. The lines in xDrip have no direct connection with data.
You can better hide root from Juggluco and hide magisk itself.

**Comment by jcarrollpe** [comment]
Thank you for a very detailed and thoughtful response. Yes I was using xDrip+ with FSL3 app for almost two weeks (my first sensor) without any compatibility issues. The (xDrip) hardware method I used was "Web follower" which requires sharing data via LibreLinkUp. So FSL3 app was connected to Libreview, and LibreLinkUp was sharing this data with xDrip+.   I switched to Juggluco because FSL3 is very sparse on features and a HUGE battery drain. I was very happy to uninstall FLS3!  I may eventually end up dropping xDrip as well, the more I discover all the features in Juggluco, and in the interest of running the least amount of apps.  

I'm sure you are already aware, but for the benefit of others I found very good [step-by-step instructions](https://github.com/maheini/FreeStyle-Libre-3-patch/blob/main/Juggluco-solution/instructions/en/instructions.md) to install Juggluco with my existing Libreview account credentials and how to properly set up xDrip as a receiver.  I followed the first 3 steps in the link. 

I was careful to allow full permissions and unrestricted battery to Juggluco, but de-selecting "Android" seems to have helped my particular connection issues. I selected it not knowing the default for an Android phone should be OFF. It seems (I guess) that when the "Glucose notification" is NOT selected, "To connect sensor" is just a 'placeholder' notification and doesn't really indicate anything problematic.  It simply allows quick access to opening the program where one can check the sensor connection status at a glance on the right side of the screen. I kept re-scanning the sensor because of the wording of the notification. I thought it was telling me to re-scan.  Lastly, I did hide Juggluco from Magisk, and hid the Magisk app itself although I have no idea why I would need to.  Abbott apps, banking apps, etc. of course but why your app requires this I haven't quite figured out... but now everything appears to be working normally and I can dig into all the smaller features you have built into your app. Thank you for your time and effort in building and maintaining this app. It is very much appreciated! Cheers.
